app/controllers/wpcc/your_details_controller.rb

```diff
@@ -24,11 +27,17 @@ def present(your_details_form)
     end
 
     def your_details_form_params
-      params.permit(:age,
-                    :salary,
-                    :gender,
-                    :salary_frequency,
-                    :employer_contribution)
+      params.require(:your_details_form).permit(:age,
+                                                :salary,
+                                                :gender,
+                                                :salary_frequency,
+                                                :contribution_preference)
+    end
+
+    def create_session
+      your_details_form_params.keys.each do |key|
```

Comment:
this is a high level question as I haven't fully checking into the code, but...
what's the reason for running `.each` on `.keys` rather than the whole params. It looks like you need to obtain the value as well anyway `params[:your_details_form][key]`...?